diff -Naur partclone-0.3.20/src/xfs/libxfs/linux.c partclone-0.3.20-patched/src/xfs/libxfs/linux.c
--- partclone-0.3.20/src/xfs/libxfs/linux.c	2022-04-27 14:56:42.000000000 +0800
+++ partclone-0.3.20-patched/src/xfs/libxfs/linux.c	2022-12-21 22:11:18.000000000 +0800
@@ -163,7 +163,7 @@
 platform_findsizes(char *path, int fd, long long *sz, int *bsz)
 {
 	struct stat	st;
-	__uint64_t	size;
+	uint64_t	size;
 	int		error;
 
 	if (fstat(fd, &st) < 0) {
diff -Naur partclone-0.3.20/src/xfs/libxfs/radix-tree.c partclone-0.3.20-patched/src/xfs/libxfs/radix-tree.c
--- partclone-0.3.20/src/xfs/libxfs/radix-tree.c	2022-04-27 14:56:42.000000000 +0800
+++ partclone-0.3.20-patched/src/xfs/libxfs/radix-tree.c	2022-12-21 22:03:46.000000000 +0800
@@ -66,21 +66,21 @@
 static inline void tag_set(struct radix_tree_node *node, unsigned int tag,
 		int offset)
 {
-	*((__uint32_t *)node->tags[tag] + (offset >> 5)) |= (1 << (offset & 31));
+	*((uint32_t *)node->tags[tag] + (offset >> 5)) |= (1 << (offset & 31));
 }
 
 static inline void tag_clear(struct radix_tree_node *node, unsigned int tag,
 		int offset)
 {
-	__uint32_t 	*p = (__uint32_t*)node->tags[tag] + (offset >> 5);
-	__uint32_t 	m = 1 << (offset & 31);
+	uint32_t 	*p = (uint32_t*)node->tags[tag] + (offset >> 5);
+	uint32_t 	m = 1 << (offset & 31);
 	*p &= ~m;
 }
 
 static inline int tag_get(struct radix_tree_node *node, unsigned int tag,
 		int offset)
 {
-	return 1 & (((const __uint32_t *)node->tags[tag])[offset >> 5] >> (offset & 31));
+	return 1 & (((const uint32_t *)node->tags[tag])[offset >> 5] >> (offset & 31));
 }
 
 /*

